### PR TITLE
Test/unit test guest and poll controller

### LINKED ISSUE #76: Create Unit test for GuestController and PollController
**GuestController**

- [X] Get all guest

- [X] Createg guest

**PollController**

- [x] Get all public polls

- [x] Get all polls to owner (as admin or owner)

- [x] Create poll

- [x] Activate poll 

- [x] Is poll activate?

- [x] Get number of votes

- [x] Delete poll (as admin or user)

- [x] Get all polls (admin)

### PR DESCRIPTION
**Pull request by oddhus**
**Added**

- Guest controller unit test (must change name to GuestControllerTest after all current PR are merged)

- Poll controller unit tests (must change name to PollControllerTest  after all current PR are merged)

- Response check class for unit controller tests

- Added builder methods to poll, guest and votesresponse

**Changed**

- Responsetype of create Poll

- Path to call getPollsByOwner from polls/polls/{id} to polls/owner/{id}

- Some documents got formatted automatically when commited 😬, therefore the lines of code change is inflated.  

Fix #76

### CODE REVIEW

File: src/main/java/no/hvl/dat250/h2020/group5/responses/VotesResponse.java [hunk 1]
```diff
@@ -6,6 +6,16 @@
...
+  Integer no;
+
+  public VotesResponse yes(int numberOfYes) {
+    this.setYes(numberOfYes);
+    return this;
+  }
+
+  public VotesResponse no(int numberOfNo) {
+    this.setNo(numberOfNo);
+    return this;
```
Comment (JennStro) [inline src/main/java/no/hvl/dat250/h2020/group5/responses/VotesResponse.java:19]: Nice 👍

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  @Test
+  public void shouldReturnOneGuestTest() throws Exception {
+    List<GuestResponse> resultList = Arrays.asList(guestResponse1, guestResponse2);
+    when(guestService.getAllGuests()).thenReturn(resultList);
+
+    mockMvc
+        .perform(MockMvcRequestBuilders.get("/guests").accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(responseBody().containsObjectAsJson(resultList));
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerUnitTest.java:55]: Dette så mer lettvint ut ja! Skal begynne å gjøre det heretter istedenfor! 💯 🥇

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    user = new User().userName("my_awesome_username");
+    user.setId(3L);
+
+    poll1 =
+        new Poll()
+            .question("my_awesome_question1")
+            .visibilityType(PollVisibilityType.PRIVATE)
+            .pollOwner(user);
+    poll1.setId(1L);
+
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerUnitTest.java:56]: Pent! 🔥

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/ResponseBodyMatchers.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  /**
+   * Method for easily checking the response in the controller unit-testing
+   *
+   * @param expectedObject, The object you expect to return from MockMvc.
+   * @return
+   */
+  public ResultMatcher containsObjectAsJson(Object expectedObject) {
+    return mvcResult -> {
+      String actualResponse = mvcResult.getResponse().getContentAsString();
+      Assertions.assertEquals(objectMapper.writeValueAsString(expectedObject), actualResponse);
```
Comment (JennStro) [inline src/test/java/no/hvl/dat250/h2020/group5/controllers/ResponseBodyMatchers.java:24]: Kjempebra! 😄

**Review by JennStro — DISMISSED** [review]
Ser veldig bra ut, flotte tester! 👍 👍

**Comment by oddhus** [comment]
Det var en merge conflict som måtte fikses 😄

**Review by JennStro — APPROVED** [review]
🎉

File: src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerUnitTest.java (hunk 2 above)
Comment (Severinzz) [reply src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerUnitTest.java:55]: Enig med @JennStro her, såg veldig mye enklere ut!

**Review by Severinzz — APPROVED** [review]
Bra jobba! 🥇

**Pull request merged**